### PR TITLE
episode data migration for DMC Lab Register

### PR DESCRIPTION
**Pull request by nickpell**
https://docs.google.com/spreadsheets/d/1ywTP_NU0yMghuUdp3LIejssmNvMdKqQ0-6GRDWJJR8I/edit#gid=0

@sheelio Doing a test run now, will share an excel file with the new case properties once the run is complete.

### CODE REVIEW

**Comment by nickpell** [comment]
cc: @proteusvacuum

File: custom/enikshay/management/commands/run_diagnosis_test_migration.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            return sorted(test_cases, key=lambda c: c.get_case_property('date_reported'))[-1]
+        else:
+            return None
+
+    @staticmethod
+    def get_updates(test):
+        test_case_properties = test.dynamic_case_properties()
+        return {
+            'diagnosis_test_result_date': test_case_properties.get('date_tested', ''),
+            'diagnosis_lab_facility_name': test_case_properties.get('testing_facility_name', ''),
```
Comment (sheelio) [inline custom/enikshay/management/commands/run_diagnosis_test_migration.py]: @nickpell 

It looks like a bunch of test cases don't have a testing_facility_name.  Can you change the migration to do something like:
`test.case_properties.get('testing_facility_name') or test.case_properties.get('testing_facility_saved_name') or test.case_properties.get('test_facility_referred_to')`

File: custom/enikshay/management/commands/run_diagnosis_test_migration.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            and case_properties.get('treatment_initiated') == 'yes_phi'
+            and case_properties.get('diagnosis_test_result_date', '') == ''
+        )
+
+    @staticmethod
+    def get_relevant_test_case(domain, episode_case):
+        try:
+            occurrence_case = get_occurrence_case_from_episode(domain, episode_case.case_id)
+        except ENikshayCaseNotFound:
+            return None
```
Comment (sheelio) [inline custom/enikshay/management/commands/run_diagnosis_test_migration.py]: @nickpell 
Is there a way to log these cases so we know about them?

File: custom/enikshay/management/commands/run_diagnosis_test_migration.py (hunk 1 above)
Comment (nickpell) [reply custom/enikshay/management/commands/run_diagnosis_test_migration.py]: (note) this is being handled here:
https://github.com/dimagi/commcare-hq/pull/17739#pullrequestreview-61254331

Comment (sheelio) [reply custom/enikshay/management/commands/run_diagnosis_test_migration.py]: Yup, no need to make this change anymore..

File: custom/enikshay/management/commands/run_diagnosis_test_migration.py (hunk 2 above)
Comment (nickpell) [reply custom/enikshay/management/commands/run_diagnosis_test_migration.py]: done

**Comment by nickpell** [comment]
@sheelio included a log file for episodes missing occurrences

**Comment by sheelio** [comment]
@nickpell 
Looks good to me.

**Pull request merged**